std/conv.d

```diff
@@ -2918,65 +3066,134 @@ if (isInputRange!Source && isSomeChar!(ElementType!Source) && !is(Source == enum
 
     enforce(!p.empty, bailOut());
 
+    static if (doCount)
+    {
+        size_t count = 0;
+    }
+
     bool sign = false;
     switch (p.front)
     {
     case '-':
         sign = true;
+        static if (doCount)
+        {
+            ++count;
+        }
         p.popFront();
         enforce(!p.empty, bailOut());
         if (toLower(p.front) == 'i')
             goto case 'i';
         break;
     case '+':
+        static if (doCount)
+        {
+            ++count;
+        }
         p.popFront();
         enforce(!p.empty, bailOut());
         break;
     case 'i': case 'I':
         // inf
+        static if (doCount)
+        {
+            ++count;
+        }
         p.popFront();
         enforce(!p.empty && toUpper(p.front) == 'N',
                bailOut("error converting input to floating point"));
+        static if (doCount)
+        {
+            ++count;
+        }
         p.popFront();
         enforce(!p.empty && toUpper(p.front) == 'F',
                bailOut("error converting input to floating point"));
         // skip past the last 'f'
+        static if (doCount)
+        {
+            ++count;
+        }
         p.popFront();
         static if (isNarrowString!Source)
             source = cast(Source) p;
-        return sign ? -Target.infinity : Target.infinity;
+        static if (doCount)
+        {
+            return tuple(sign ? -Target.infinity : Target.infinity, count);
```

Comment:
... and then conditionally return the tuple (or not, if doCount is false), rather than have `static if (doCount)` all over the place. Ditto throughout.

Reply:
Thank you for the feedback! I have a question, are these last 3 comments regarding only this overload, or all the overloads?

Reply:
All of the overloads, sorry if that wasn't clear from `Ditto throughout`.

Reply:
Ok, thank you for the clarification!